File: Authorize/src/main/java/com/experiment/authorize/repository/base/EntityRepository.java
package com.experiment.authorize.repository.base;

import com.experiment.authorize.entity.base.BaseEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

@Repository
public interface EntityRepository<T extends BaseEntity>  extends JpaRepository<T, Long>, JpaSpecificationExecutor<T>{

}

File: Inventory/src/main/kotlin/com/experiment/inventory/repository/EntityRepository.kt
package com.experiment.inventory.repository

import com.experiment.inventory.entity.base.BaseEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository

@Repository
interface EntityRepository<T>: JpaRepository<T, Long>, JpaSpecificationExecutor<T> where T: BaseEntity {
}